lib/vega_lite/viewer.ex

```diff
@@ -0,0 +1,45 @@
+defmodule VegaLite.Viewer do
+  @moduledoc """
+  Graphics rendering using Erlang's `:wx` bindings.
+
+  This method if useful for viewing graphics in scripts
+  and iex sessions.
+  """
+
+  @doc """
+  Renders a `VegaLite` specification in GUI window widget.
+
+  Requires Erlang compilation to include the `:wx` module.
+  """
+  @spec show(VegaLite.t()) :: :ok | :error
+  def show(vl) do
+    with {:ok, _pid} <- start_wx_viewer(vl), do: :ok
+  end
+
+  @doc """
+  Same as `show/1`, but blocks until the window widget is closed.
+  """
+  @spec show_and_wait(VegaLite.t()) :: :ok | :error
+  def show_and_wait(vl) do
+    with {:ok, pid} <- start_wx_viewer(vl) do
+      ref = Process.monitor(pid)
+
+      receive do
+        {:DOWN, ^ref, :process, _object, _reason} -> :ok
+      end
+    end
+  end
+
+  if Code.ensure_compiled(:wx) == {:module, :wx} do
```

Comment:
@josevalim the `:wx` module is available or not depending on how Erlang was compiled, so to avoid compilation warnings I similarly made our compilation conditional here and in the `WxViewer` module. Happy to hear if there's any better way :)

Reply:
Alternatively we could always define `WxViewer` but define it like:

```elixir
defmodule VegaLite.WxViewer do
  if Code.ensure_compiled(:wx) == {:module, :wx} do
    def start do # throw
  else
    @behaviour :wx_object

    def start do #...

    # ...
  end
end
```

Reply:
Yes, I prefer the second version, so everything is encapsulated in a single place.